data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/coordination/state/ExportProgressState.java

```diff
@@ -32,6 +33,12 @@ public class ExportProgressState {
     @JsonProperty("tables")
     private List<String> tables;
 
+    /**
+     * Map of table name to primary keys
+     */
+    @JsonProperty("primaryKeyMap")
+    private Map<String, List<String>> primaryKeyMap;
```

Comment:
Does it take into account for composite keys ? Primary is key a combination of multiple columns.

Reply:
Yes, it's a List of strings and will be used here to form the document_id: https://github.com/opensearch-project/data-prepper/blob/642db0d49bb2ac9581c1c58602d9417cec1eb9d8/data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/converter/ExportRecordConverter.java#L46-L48